noise/conn.go

```diff
@@ -351,6 +355,21 @@ func (c *Conn) IsRemoteAuthenticated() bool {
 	return c.isRemoteAuthenticated
 }
 
+// StaticKey returns the static key of the remote peer. It is useful in case the
+// static key is only transmitted during the handshake.
+func (c *Conn) StaticKey() ([]byte, error) {
+	if !c.IsRemoteAuthenticated() {
+		return nil, errors.New("noise: remote peer not authenticated")
+	}
+	if !c.handshakeComplete {
+		return nil, errors.New("noise: handshake not completed")
+	}
+	if c.remotePub == nil {
+		return nil, errors.New("noise: no remote static key given")
+	}
+	return c.remotePub, nil
```

Comment:
why not returning `hs.rs.PublicKey` instead? This way you wouldn't have to create a `c.remotePub` and copy the public key into it.

Reply:
Because the `hs.rs` is cleared when the handshake completes (`c.hs.clear()`) and the clearing clears up also the public key. I think it could be relaxed to only clear up the private key but better be safe than sorry.